File: go/examples/advanced.go
//go:build ignore
// +build ignore

package main

import (
	"time"

	tracing "github.com/LeDuyViet/quicktrace/go"
)

func main() {
	// Advanced usage với smart filtering
	tracer := tracing.NewSimpleTracer("Advanced Example",
		tracing.WithOutputStyle(tracing.StyleDetailed),
		tracing.WithHideUltraFast(1*time.Millisecond),
		tracing.WithShowSlowOnly(10*time.Millisecond),
		tracing.WithGroupSimilar(5*time.Millisecond),
	)

	// Simulate database operations
	time.Sleep(100 * time.Millisecond)
	tracer.Span("Connect to database")

	time.Sleep(45 * time.Millisecond)
	tracer.Span("Execute query 1")

	time.Sleep(50 * time.Millisecond) // Similar to query 1
	tracer.Span("Execute query 2")

	time.Sleep(5 * time.Millisecond)
	tracer.Span("Cache result")

	time.Sleep(500 * time.Microsecond) // Will be hidden
	tracer.Span("Ultra fast operation")

	time.Sleep(200 * time.Millisecond)
	tracer.Span("Process business logic")

	time.Sleep(30 * time.Millisecond)
	tracer.Span("Send notification")

	tracer.End()
}

File: js/examples/advanced.js
const { Tracer } = require('../tracer.js');

async function main() {
    // Advanced usage với smart filtering
    const tracer = new Tracer('Advanced Example', {
        style: 'detailed',
        hideUltraFast: 1,      // ms
        showSlowOnly: 10,      // ms  
        groupSimilar: 5,       // ms
        minTotalDuration: 50   // ms
    });

    // Simulate database operations
    await new Promise(resolve => setTimeout(resolve, 100));
    tracer.span('Connect to database');

    await new Promise(resolve => setTimeout(resolve, 45));
    tracer.span('Execute query 1');

    await new Promise(resolve => setTimeout(resolve, 50)); // Similar to query 1
    tracer.span('Execute query 2');

    await new Promise(resolve => setTimeout(resolve, 5));
    tracer.span('Cache result');

    await new Promise(resolve => setTimeout(resolve, 0.5)); // Will be hidden
    tracer.span('Ultra fast operation');

    await new Promise(resolve => setTimeout(resolve, 200));
    tracer.span('Process business logic');

    await new Promise(resolve => setTimeout(resolve, 30));
    tracer.span('Send notification');

    tracer.end();
}

main().catch(console.error); 